source/common/http/conn_manager_utility.cc

```diff
@@ -127,6 +127,7 @@ Network::Address::InstanceConstSharedPtr ConnectionManagerUtility::mutateRequest
       request_headers.removeEnvoyDownstreamServiceNode();
     }
 
+    request_headers.removeEnvoyRetriableStatusCodes();
```

Comment:
Is this behavior validated in tests?

Reply:
Maybe I'm missing it, but where do you check the header has been removed?

Reply:
On the line that I linked. Removing `request_headers.removeEnvoyRetriableStatusCodes();` fails this test:
```
 TEST_F(ConnectionManagerUtilityTest, ExternalAddressExternalRequestUseRemote) {
   connection_.remote_address_ = std::make_shared<Network::Address::Ipv4Instance>("50.0.0.1");
   ON_CALL(config_, useRemoteAddress()).WillByDefault(Return(true));
  route_config_.internal_only_headers_.push_back(LowerCaseString("custom_header"));
  TestHeaderMapImpl headers{{"x-envoy-decorator-operation", "foo"},
                            {"x-envoy-downstream-service-cluster", "foo"},
                            {"x-envoy-retriable-status-codes", "123,456"},
                            {"x-envoy-retry-on", "foo"},
                            {"x-envoy-retry-grpc-on", "foo"},
                            {"x-envoy-max-retries", "foo"},
                             {"x-envoy-upstream-alt-stat-name", "foo"},
                             {"x-envoy-upstream-rq-timeout-alt-response", "204"},
                             {"x-envoy-upstream-rq-timeout-ms", "foo"},
                             {"x-envoy-expected-rq-timeout-ms", "10"},
                             {"x-envoy-ip-tags", "bar"},
                             {"custom_header", "foo"}};
 
   EXPECT_EQ((MutateRequestRet{"50.0.0.1:0", false}),
             callMutateRequestHeaders(headers, Protocol::Http2));
  EXPECT_EQ("50.0.0.1", headers.get_("x-envoy-external-address"));
  EXPECT_FALSE(headers.has("x-envoy-decorator-operation"));
  EXPECT_FALSE(headers.has("x-envoy-downstream-service-cluster"));
  EXPECT_FALSE(headers.has("x-envoy-retriable-status-codes")); // THIS ASSERT RIGHT HERE
  EXPECT_FALSE(headers.has("x-envoy-retry-on"));
  EXPECT_FALSE(headers.has("x-envoy-retry-grpc-on"));
  EXPECT_FALSE(headers.has("x-envoy-max-retries"));
   EXPECT_FALSE(headers.has("x-envoy-upstream-alt-stat-name"));
   EXPECT_FALSE(headers.has("x-envoy-upstream-rq-timeout-alt-response"));
   EXPECT_FALSE(headers.has("x-envoy-upstream-rq-timeout-ms"));
   EXPECT_FALSE(headers.has("x-envoy-expected-rq-timeout-ms"));
   EXPECT_FALSE(headers.has("x-envoy-ip-tags"));
   EXPECT_FALSE(headers.has("custom_header"));
 }
```

Reply:
Ah, sorry, missed that before. Thanks, LGTM to merge when CI passes.